docs/reference/module-api/module-overview.md

```diff
@@ -483,18 +483,354 @@ module.exports = {
+| `'conflict'` | 409 |
+| `'locked'` | 409 |
+| `'unprocessable'` | 422 |
+| `'unimplemented'` | 501 |
+
+Passing a different value as the first argument to `self.apos.error()` will set the response code to 500.
+<!-- TODO: Link to the method's own documentation page when available for more. -->
+
+#### `extendApiRoutes(self)`
+
+Extend the behavior of existing API routes (set in `apiRoutes`) in the `extendApiRoutes` section. This function must return an object as described in [`apiRoutes`](#apiroutes-self).
+
+Each extended API route function should accept the original function as `_super` and the `req` request object. They should return data in a similar format to the existing API route.
+
+```javascript
+// modules/featured-product/index.js
+module.exports = {
+  extend: 'product',
+  // ...
+  extendApiRoutes(self) {
+    return {
+      get: {
+        // GET /api/v1/featured-product/cheapest
+        async cheapest(_super, req) {
+          const response = _super(req);
+
+          // Update the response object...
+
+          return response;
+        }
+      }
+    };
+  }
+};
+```
+
 ### `renderRoutes(self)`
+
+Add custom API routes to return rendered templates. The `apiRoutes` function takes takes the module as an argument and must return an object with properties for the relevant [HTTP request method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods), including `get`, `post`, `patch`, and `delete`. Each of those properties should be set to an object of functions.
+
+The name of the route dictates the template file that will be rendered. For example the `latest` route in the `product` module's `renderRoutes` section will return the template at `/modules/product/views/latest.html`.
+
+Information returned by the component function will be used in the associated template as `data`.
+
+```javascript
+// modules/product/index.js
+module.exports = {
+  // ...
+  renderRoutes(self) {
+    return {
+      get: {
+        // GET /api/v1/product/latest
+        // The route rendered HTML for /modules/product/views/latest.html
+        async latest(req) {
+          const products = await self.find(req)
+            .sort({ createdAt: -1 })
+            .limit(req.query.max || 5)
+            .toArray();
+
+          return {
+            products
+          };
+        }
+      }
+    };
+  }
+};
+```
+
 ### `routes(self)`
+
+Add standard Express routes. The `routes` function takes takes the module as an argument and must return an object with properties for the relevant [HTTP request method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods), including `get`, `post`, `patch`, and `delete`. Each of those properties should be set to an object of functions.
+
+Each `routes` function takes the Express arguments `req` (the [request object](https://expressjs.com/en/api.html#req)) and `res` (the [response object](https://expressjs.com/en/api.html#res)). The functions must return using the `res.redirect` or `res.send` methods.
```

Comment:
How about: "the functions must generate a response via `res` to avoid leaking resources, typically using the `res.redirect` or `res.send` methods." There are other, less often used ways to end a response, such as calling `res.send`, so I don't want to make it sound like Apostrophe is doing anything other than letting you do as you like with Express here.

Reply:
`routes` should be used only where `apiRoutes` or `restApiRoutes` or `renderRoutes` won't do, such as when you need to use `res.redirect` or pipe a stream. We should state this clearly to help guide people to the painless solution. Failing to do the right thing with `res` or handle errors etc. is a common cause of Node.js bugs, which is why apiRoutes exists.

(At some point I'd like to add a way to use `res.redirect` via a return value. Piping a stream will probably always be an "uh... just use routes()" scenario.")

Reply:
I've added a new :::tip block a few lines down.